cmd/public-ip/main.go

```diff
@@ -10,14 +11,27 @@ import (
 )
 
 func main() {
+	// Url to get IP info
+	var urlForIp string = "https://ifconfig.co/ip"
+	var urlForCity string = "https://ifconfig.co/city"
 
 	// Set display texts to defaults.
 	var fullText string = "unknown"
 	var shortText string = "unknown"
+	var whatIsMyIpUrl string
+
+	var cityFlag = flag.Bool("city", false, "pass -city to report City instead of IP")
+	flag.Parse()
+
+	if cityFlag != nil && *cityFlag {
+		whatIsMyIpUrl = urlForCity
+	} else {
+		whatIsMyIpUrl = urlForIp
+	}
 
 	// Request whats-my-ip service to return this
 	// machine's public IP address via TLS.
-	resp, err := http.Get("https://ip.wirelab.org/")
+	resp, err := http.Get(whatIsMyIpUrl)
```

Comment:
This is a really good suggestion, I should have changed this (even make the URL to query configurable) a long time ago!

Reply:
Thanks